packages/logs/src/rawLogsEvent.types.ts

```diff
@@ -1,29 +1,67 @@
 import type { Context, ErrorSource, TimeStamp } from '@datadog/browser-core'
 import type { StatusType } from './domain/logger'
 
-export interface RawLogsEvent {
-  date?: TimeStamp
+export type RawLogsEvent =
+  | RawConsoleLogsEvent
+  | RawNetworkLogsEvent
+  | RawLoggerLogsEvent
+  | RawAgentLogsEvent
+  | RawReportLogsEvent
+  | RawRuntimeLogsEvent
+
+export type RawLogsEventOrigin = RawLogsEvent['origin']
+
+type Error = {
+  kind?: string
+  origin: ErrorSource // Todo: Remove in the next major release
+  stack?: string
+  [k: string]: unknown
+}
+
+interface CommonRawLogsEvent {
   message: string
   status: StatusType
-  origin: ErrorSource
-  logger?: {
-    name: string
-    [k: string]: unknown
-  }
-  error?: {
-    kind?: string
-    origin: ErrorSource
-    stack?: string
-    [k: string]: unknown
-  }
-  http?: {
+  error?: Error
+}
+
+interface CommonRawErrorLogsEvent extends CommonRawLogsEvent {
+  status: typeof StatusType.error
+  error: Error
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏personal preference, I would inline this interface to avoid to navigating across the hierarchy to understand what we actually have in an interface